### ISSUE TITLE
vmware_guest_snapshot: Idempotency issue

### ISSUE THREAD

**Issue by dagwieers** [ansible/ansible#24511]
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME

vmware_guest_snapshot

##### ANSIBLE VERSION

v2.3+

##### SUMMARY

So when you run a basic vmware_guest_snapshot multiple times with the same snapshot name, it creates a snapshot onto the existing snapshot, and not report the snapshot is present.

So after 2 runs, you get this:

    virtual machine
        snapshot001
            snapshot001

instead of this:

    virtual machine
        snapshot001

And both runs show up as being changed.

The interface probably need more information to do the right thing, both for creation as well as removal. You probably need to point your parent as well ?

**Comment by ansibot** [comment]
cc @nerzhul
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by Tronde** [comment]
Hi, I can confirm this issue. For better idempotency I would suggest to fail if a snapshot with a given name already exists.

I just tried what happens when I run my playbook a second time and after that going to revert the snapshot. After the second run another snapshot with the same name was created. Now, I changed my playbook to revert the snapshot. The VM was not reverted to the last taken snapshot with the given name but to the first one.

The module couldn't decide which snapshot to revert, because they have the same name. To prevent this from happening and to keep it as simple and stupid as possible I would strongly recommend to not allow the creation of a second snapshot with an identical name.

Best regards,
Joerg
